src/Symfony/Component/Messenger/Exception/ValidationFailedException.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Exception;
+
+use Symfony\Component\Validator\ConstraintViolationListInterface;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class ValidationFailedException extends \RuntimeException implements ExceptionInterface
+{
+    protected $violations;
+    protected $violatingMessage;
+
+    public function __construct($violatingMessage, ConstraintViolationListInterface $violations)
+    {
+        $this->violatingMessage = $violatingMessage;
+        $this->violations = $violations;
+
+        parent::__construct($this->__toString());
+    }
+
+    public function __toString()
```

Comment:
AFAIK, we never defined `__toString` method on exceptions in the core. Not sure this is desired now.